clippy_lints/src/types.rs

```diff
@@ -1449,3 +1450,326 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for InvalidUpcastComparisons {
         }
     }
 }
+
+/// **What it does:** Checks for `impl` or `fn` missing generalization over
+/// different hashers and implicitly defaulting to the default hashing
+/// algorithm (SipHash). This lint ignores private free-functions.
+///
+/// **Why is this bad?** `HashMap` or `HashSet` with custom hashers cannot be
+/// used with them.
+///
+/// **Known problems:** Suggestions for replacing constructors are not always
+/// accurate.
+///
+/// **Example:**
+/// ```rust
+/// impl<K: Hash + Eq, V> Serialize for HashMap<K, V> { ... }
+///
+/// pub foo(map: &mut HashMap<i32, i32>) { .. }
+/// ```
+declare_lint! {
+    pub IMPLICIT_HASHER,
+    Warn,
+    "missing generalization over different hashers"
+}
+
+pub struct ImplicitHasher;
+
+impl LintPass for ImplicitHasher {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(IMPLICIT_HASHER)
+    }
+}
+
+impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ImplicitHasher {
+    #[allow(cast_possible_truncation)]
+    fn check_item(&mut self, cx: &LateContext<'a, 'tcx>, item: &'tcx Item) {
+        if let ItemImpl(_, _, _, ref generics, _, ref ty, ref items) = item.node {
+            let mut vis = ImplicitHasherTypeVisitor::new(cx);
+            vis.visit_ty(ty);
+
+            for target in vis.found {
+                let generics_snip = snippet(cx, generics.span, "");
+                let generics_snip_trimmed = if generics_snip.len() == 0 {
```

Comment:
you can skip this check if you make the last argument of `snippet` be `"<>"`

Reply:
`snippet(cx, generics.span, "<>")` seems to return an empty string when there is no generics.